src/neo4j_genai/schema.py

```diff
@@ -0,0 +1,138 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from typing import Any
+
+import neo4j
+
+
+BASE_ENTITY_LABEL = "__Entity__"
+EXCLUDED_LABELS = ["_Bloom_Perspective_", "_Bloom_Scene_"]
+EXCLUDED_RELS = ["_Bloom_HAS_SCENE_"]
+
+NODE_PROPERTIES_QUERY = """
+CALL apoc.meta.data()
+YIELD label, other, elementType, type, property
+WHERE NOT type = "RELATIONSHIP" AND elementType = "node" 
+    AND NOT label IN $EXCLUDED_LABELS
+WITH label AS nodeLabels, collect({property:property, type:type}) AS properties
+RETURN {labels: nodeLabels, properties: properties} AS output
+"""
+
+REL_PROPERTIES_QUERY = """
+CALL apoc.meta.data()
+YIELD label, other, elementType, type, property
+WHERE NOT type = "RELATIONSHIP" AND elementType = "relationship"
+    AND NOT label in $EXCLUDED_LABELS
+WITH label AS nodeLabels, collect({property:property, type:type}) AS properties
+RETURN {type: nodeLabels, properties: properties} AS output
+"""
+
+REL_QUERY = """
+CALL apoc.meta.data()
+YIELD label, other, elementType, type, property
+WHERE type = "RELATIONSHIP" AND elementType = "node"
+UNWIND other AS other_node
+WITH * WHERE NOT label IN $EXCLUDED_LABELS
+    AND NOT other_node IN $EXCLUDED_LABELS
+RETURN {start: label, type: property, end: toString(other_node)} AS output
+"""
+
+
+def query_database(
+    driver: neo4j.Driver, query: str, params: dict = {}
+) -> list[dict[str, Any]]:
+    """
+    Queries the database.
+
+    Args:
+        driver (neo4j.Driver):  Neo4j Python driver instance.
+        query (str): The cypher query.
+        params (dict, optional): The query parameters. Defaults to {}.
+
+    Returns:
+        List[Dict[str, Any]]: the result of the query in json format.
+    """
```

Comment:
A weird quirk of Python is that passing a mutable dictionary as a default argument to a function can have unforeseen consequences, [see this stackoverflow post](https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python).

```suggestion
def query_database(
    driver: neo4j.Driver, query: str, params: Optional[dict] = None
) -> list[dict[str, Any]]:
    """
    Queries the database.

    Args:
        driver (neo4j.Driver):  Neo4j Python driver instance.
        query (str): The cypher query.
        params (dict, optional): The query parameters. Defaults to {}.

    Returns:
        List[Dict[str, Any]]: the result of the query in json format.
    """
    if params is None:
        params = {}
```

Reply:
[8faea36](https://github.com/neo4j/neo4j-genai-python/pull/32/commits/8faea365b5fa532ddc3d461e95b4c30d4614608d)